patches/0012-remove-long-path-support-hack.patch

```diff
@@ -0,0 +1,544 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: qmuntal <quimmuntal@gmail.com>
+Date: Tue, 12 Mar 2024 12:26:53 +0100
+Subject: [PATCH] remove long path support hack
+
+This commit backports https://github.com/microsoft/go/pull/1171,
+https://go-review.googlesource.com/c/go/+/570995, and
+https://go-review.googlesource.com/c/go/+/574695 to the Go 1.21
```

Comment:
I feel like it would be easier to review these (and maintain merge conflicts in the future) if each port were in its own patch. The whole `github.com/microsoft/go-infra/cmd/git-go-patch command: patch number 100` thing could help keep context loosely together.

(I'm a bit late with this realization for now, can do it later (or only in the future) if it makes sense.)

Reply:
Makes sense, I'll do, thanks!